R/archive.R

```diff
@@ -439,69 +439,6 @@ epi_archive <-
         self$clobberable_versions_start <- clobberable_versions_start
         self$versions_end <- versions_end
       },
-      #' Print information about an archive
-      #' @param class Boolean; whether to print the class label header
-      #' @param methods Boolean; whether to print all available methods of
-      #'   the archive
-      print = function(class = TRUE, methods = TRUE) {
-        if (class) cat("An `epi_archive` object, with metadata:\n")
-        cat(sprintf("* %-9s = %s\n", "geo_type", self$geo_type))
-        cat(sprintf("* %-9s = %s\n", "time_type", self$time_type))
-        if (!is.null(self$additional_metadata)) {
-          sapply(self$additional_metadata, function(m) {
-            cat(sprintf("* %-9s = %s\n", names(m), m))
-          })
-        }
-        cat("----------\n")
-        if (length(self$DT$time_value) == 0 || all(is.na(self$DT$time_value))) {
-          min_time <- max_time <- NA
-        } else {
-          min_time <- Min(self$DT$time_value)
-          max_time <- Max(self$DT$time_value)
-        }
-        cat(sprintf("* %-14s = %s\n", "min time value", min_time))
-        cat(sprintf("* %-14s = %s\n", "max time value", max_time))
-        cat(sprintf(
-          "* %-14s = %s\n", "first version with update",
-          min(self$DT$version)
-        ))
-        cat(sprintf(
-          "* %-14s = %s\n", "last version with update",
-          max(self$DT$version)
-        ))
-        if (is.na(self$clobberable_versions_start)) {
-          cat("* No clobberable versions\n")
-        } else {
-          cat(sprintf(
-            "* %-14s = %s\n", "clobberable versions start",
-            self$clobberable_versions_start
-          ))
-        }
-        cat(sprintf(
-          "* %-14s = %s\n", "versions end",
-          self$versions_end
-        ))
-        cat("----------\n")
-        cat(sprintf(
-          "Data archive (stored in DT field): %i x %i\n",
-          nrow(self$DT), ncol(self$DT)
-        ))
-        cat(sprintf("Columns in DT: %s\n", paste(ifelse(length(
-          colnames(self$DT)
-        ) <= 4, paste(colnames(self$DT), collapse = ", "),
-        paste(
-          paste(colnames(self$DT)[1:4], collapse = ", "), "and",
-          length(colnames(self$DT)[5:length(colnames(self$DT))]), "more columns"
-        )
-        ))))
-        if (methods) {
-          cat("----------\n")
-          writeLines(wrap_varnames(
-            initial = "Public R6 methods: ",
-            names(epi_archive$public_methods)
-          ))
-        }
-      },
```

Comment:
It's fine but unnecessary to move this from R6 -> S3 at this point.  (R6 takes care of providing a `print` S3 method forwarding appropriately.  If we were planning to stick with R6, we might want it here so we could potentially change some things into private members & still use them in the print implementation.  Users could also use `x$print()`, thought that's likely not in high demand.)

Reply:
Got it, this is something we want to do eventually though right? Does it make sense to do the R6 -> S3 migration incrementally or should it be more discrete?

Reply:
Eventually, yes.  It'd be nice to distribute the effort incrementally, but (i) it doesn't make sense for users to see a partial migration so it might interfere with releases, and (ii) the complete migration may involve some renaming and access pattern changes that might require scanning ALL of the archive code, so we might have to revisit "already-migrated" stuff.

Reply:
Alright, I'll roll it back then

Reply:
issue: `grouped_epi_archive.R` still uses `$print`:

```
        private$ungrouped$print(class = FALSE, methods = FALSE)
```